### PR TITLE
State archival on the client side

### LINKED ISSUE #596: Create a guide or guides for dapp developers who are looking to implement state archival on the client side
**- Get awarded:** up to $700 worth of XLM
**- Section:** How-To Guides - Dapp Development: https://github.com/stellar/stellar-docs/tree/main/docs/smart-contracts/guides/dapps

**- Guide(s) could include:**
  - Introduction to state archival:
    - Define state archival in the context of Stellar dapps
    - Explain why managing state archival is important for applications
    - Describe various methods to implement state archival on the client side
  - Practical examples:
    - Provide code examples showing how to set up and manage state archival
    - Include scenarios where different techniques might be more appropriate
  - Challenges and solutions:
    - Identify common challenges developers might face with state archival
    - Offer solutions to these challenges

### PR DESCRIPTION
**Pull request by Myestery**
fix #596

### CODE REVIEW

**Comment by briwylde08** [comment]
Thank you! We'll get this assigned and reviewed!

**Comment by Myestery** [comment]
Hello Bri
Kindly see this PR, it hasn't been assigned too

**Comment by briwylde08** [comment]
Hi @Myestery, apologies, we are a little behind on reviews. I'll get it assigned next week!

**Comment by Myestery** [comment]
Little reminder @briwylde08 that this PR hasn't been assigned yet

**Comment by briwylde08** [comment]
Hi @Myestery! I know @Shaptic is aware of this one, review coming soon.

File: docs/build/guides/dapps/state-archival.mdx
Comment (Shaptic) [inline docs/build/guides/dapps/state-archival.mdx:46]: Isn't everything up to this point already described at length in your [cross reference](https://developers.stellar.org/docs/learn/encyclopedia/storage/state-archival)? It'd be more valuable if we focused on the client side instead of repeating things.

File: docs/build/guides/dapps/state-archival.mdx
Comment (Shaptic) [inline docs/build/guides/dapps/state-archival.mdx:61]:
```suggestion
This method involves invoking [the `extend_ttl()` method](https://docs.rs/soroban-sdk/latest/soroban_sdk/storage/struct.Instance.html#method.extend_ttl) from your smart contract to extend the TTL of the contract instance and its associated data. This method is useful when you want to keep the data accessible for a longer period. To use this method, your contract must not be archived at the time of calling.
```

File: docs/build/guides/dapps/state-archival.mdx
Comment (Shaptic) [inline docs/build/guides/dapps/state-archival.mdx:128]: This sounds like it should be in a step 2 section?

File: docs/build/guides/dapps/state-archival.mdx
Comment (Shaptic) [inline docs/build/guides/dapps/state-archival.mdx:124]: Generally speaking, TTL extension should be a normal part of the regular operations that people make. So you wouldn't call extend_temporary on its own, but rather would call it as a normal part of some "dao_mint" or something like that. It'd be good if the examples demonstrated that at least in part.

File: docs/build/guides/dapps/state-archival.mdx
Comment (Shaptic) [inline docs/build/guides/dapps/state-archival.mdx:371]:
```suggestion
    // Fetch the WASM entry from the ledger
```

File: docs/build/guides/dapps/state-archival.mdx
Comment (Shaptic) [inline docs/build/guides/dapps/state-archival.mdx:421]: It might make this article shorter with less repeats if you cross-referenced this tutorial for this portion: https://developers.stellar.org/docs/data/rpc/api-reference/methods/getLedgerEntries#requesting-a-contracts-wasm-code

**Review by Shaptic — COMMENTED** [review]
Sorry for the delay in reviews; with Protocol 22 coming we have a pretty intense schedule crunch. I've finally finished the first-pass review! This guide looks good, most of my critiques involve the fact that a lot of these could be delegated to cross-references to focus on the extra stuff this article is adding.

**Comment by briwylde08** [comment]
Hi @Myestery! Just checking in on this one. Are you able to implement the suggested changes?

File: docs/build/guides/dapps/state-archival.mdx
Comment (Shaptic) [inline docs/build/guides/dapps/state-archival.mdx:280]: nit: probably don't need this!
```suggestion
```

File: docs/build/guides/dapps/state-archival.mdx
Comment (Shaptic) [inline docs/build/guides/dapps/state-archival.mdx:259]: It'd be good to cross-reference the [existing example](https://developers.stellar.org/docs/data/rpc/api-reference/methods/getLedgerEntries#requesting-a-contracts-wasm-code) here so users can learn more!

**Review by Shaptic — APPROVED** [review]
Great work @Myestery! I just left two tiny comments and otherwise think this is really useful content that can be merged :+1: please resolve them and I'll let @briwylde08 hit the merge button once that's done :)

**Comment by Shaptic** [comment]
@briwylde08 this is good to go! Thank you again @Myestery :)

**Comment by briwylde08** [comment]
Thank you both so much!

**Pull request merged**
